jwst/tests_nightly/general/miri/test_miri_steps_single.py

```diff
@@ -287,3 +286,110 @@ def test_miri_setpointing(self):
+        sci_cal_1d = Extract1dStep.call(input_sci_cal_file)
+
+        sci_cal_1d_data = sci_cal_1d.spec[0].spec_table['flux']
+        result_1d_data = result_1d.spec[0].spec_table['flux']
+        sci_wave = sci_cal_1d.spec[0].spec_table['wavelength']
+        # find the valid wavelenth of the user provided spectrun
+        user_wave = input_1d_bkg_model.spec[0].spec_table['wavelength']
+        user_flux = input_1d_bkg_model.spec[0].spec_table['flux']
+        user_wave_valid = np.where(user_flux > 0)
+        min_user_wave = np.amin(user_wave[user_wave_valid])
+        max_user_wave = np.amax(user_wave[user_wave_valid])
+        input_1d_bkg_model.close()
+
+        # find the waverange covered by both user and science                                          
+        sci_wave_valid = np.where(sci_cal_1d_data > 0)
+        min_wave = np.amin(sci_wave[sci_wave_valid])
+        max_wave = np.amax(sci_wave[sci_wave_valid])
+        if min_user_wave > min_wave: min_wave = min_user_wave
+        if max_user_wave < max_wave: max_wave = max_user_wave
+
+        # Compare the data 
+        sub_spec = sci_cal_1d_data - result_1d_data
+        valid = np.where(np.logical_and(sci_wave > min_wave, sci_wave < max_wave))
+        sub_spec = sub_spec[valid]
+        mean_sub = np.absolute(np.nanmean(sub_spec))
+        atol = 0.00005
+        rtol = 0.000001
+        assert_allclose(mean_sub, 0, atol=atol, rtol=rtol)
+        # _____________________________________________________________________
+        # Test 2
+        # Compare result (background subtracted image) to science image with no
+        # background. Subtract these images, smooth the subtracted image and
+        # the mean should be close to zero.
+
+        input_sci = datamodels.open(input_sci_cal_file)
+
+        # find the LRS region
+        bb = result.meta.wcs.bounding_box
+        x, y = grid_from_bounding_box(bb)
+        result_lrs_region = result.data[y.astype(int), x.astype(int)]
+        sci_lrs_region = input_sci.data[y.astype(int), x.astype(int)]
+
+        # do a 5 sigma clip on the science image
+        sci_mean = np.nanmean(sci_lrs_region)
+        sci_std = np.nanstd(sci_lrs_region)
+        upper = sci_mean + sci_std*5.0
+        lower = sci_mean - sci_std*5.0
+        mask_clean = np.logical_and(sci_lrs_region < upper, sci_lrs_region > lower)
+
+        sub = result_lrs_region - sci_lrs_region
+        mean_sub = np.absolute(np.mean(sub[mask_clean]))
+
+        atol = 0.5
+        rtol = 0.001
+        assert_allclose(mean_sub, 0, atol=atol, rtol=rtol)
+# ______________________________________________________________________
+        # Test 3 Compare background subtracted science data (results)
+        #  to a truth file.
+
+        ref_file = self.get_data(*self.ref_loc,
+                                  'miri_lrs_sci+bkg_masterbackgroundstep.fits')
+
+        result_file = result.meta.filename
+        result.save(result_file)
```

Comment:
The above 2 lines are actually overwriting the input as `result.meta.filename` still contains the input filename.  Instead, use the `save_results=True` arg to the step as mentioned further up, the call to `save()` above.

Also, could you rename `ref_file` to `truth_file` throughout these tests?  It makes it clear that these are truth files, not CRDS reference files.